### ISSUE TITLE
Do not enclose str items in quotes in HTML repr

### ISSUE THREAD

**Issue by cbrnr** [pola-rs/polars#18102]
### Description

The normal `pl.DataFrame` repr shows items in a `str` column without quotes, which I think is a good idea because (1) it takes up less space and (2) the column type is always shown anyway (and (3) it is consistent with how other packages visualize string items).

```python
import polars as pl

df = pl.DataFrame({
    "a": [1, 2, 3, 4, 5],
    "b": ["A", "B", "C", "D", "E"],
    "c": [1.1, -2.8, 3.4, 4.7, -5.9],
})

df
```

```
shape: (5, 3)
┌─────┬─────┬──────┐
│ a   ┆ b   ┆ c    │
│ --- ┆ --- ┆ ---  │
│ i64 ┆ str ┆ f64  │
╞═════╪═════╪══════╡
│ 1   ┆ A   ┆ 1.1  │
│ 2   ┆ B   ┆ -2.8 │
│ 3   ┆ C   ┆ 3.4  │
│ 4   ┆ D   ┆ 4.7  │
│ 5   ┆ E   ┆ -5.9 │
└─────┴─────┴──────┘
```

However, the HTML representation in a Jupyter notebook wraps every str item in double quotes:

<img width="123" alt="Screenshot 2024-08-08 at 16 42 08" src="https://github.com/user-attachments/assets/6679519d-ef4c-4752-bab9-7ede1e439b56">

This is inconsistent with the normal repr and arguably worse for the reasons mentioned above. Therefore, I suggest to not wrap string items in quotes in the HTML repr as well.

**Comment by mcrumiller** [comment]
Strongly agree that not having quotes is visually more appealing.

**Comment by deanm0000** [comment]
I agree that it's visually more appealing BUT I'm so used to them now that I find them nice. One example is wanting to filter by some value so copy/paste includes the quotes which is just a slight convenience.

As an aside, if you want to make it look nice for consumption then try great_tables.

**Comment by cbrnr** [comment]
That's a great package! However, this issue is not about creating publication-quality tables, but providing a consistent¹ and visually appealing² default HTML representation. Sure it can be convenient in some cases, but I'd think that these are very likely less frequently encountered than the reasons that speak for removing them.

¹consistent not only with the normal Polars repr, but also with other packages like pandas, DataFrames.jl, Tibble, etc.
²visually appealing also implies making it easy for users to extract relevant information from the output, and quotes around every string item certainly make it harder to do that IMO

**Comment by cbrnr** [comment]
PS: Maybe a compromise here could be to include quotes for a Series, but to not have them in a DataFrame?

**Comment by etiennebacher** [comment]
I don't have a strong opinion on this, but I just want to mention that quotation marks are useful to distinguish strings that have only whitespaces but a different number of them. For example, if you remove quotation marks, can you still distinguish `""` and `"   "` in the HTML repr?

For reference, the `tidyverse` print methods in R only adds quotation marks when there is at least one empty string:

``` r
tibble::tibble(a = c("a", "b", "c"))
#> # A tibble: 3 × 1
#>   a    
#>   <chr>
#> 1 a    
#> 2 b    
#> 3 c
tibble::tibble(a = c("", "  ", "a"))
#> # A tibble: 3 × 1
#>   a    
#>   <chr>
#> 1 ""   
#> 2 "  " 
#> 3 "a"
tibble::tibble(a = c("", "", "a"))
#> # A tibble: 3 × 1
#>   a    
#>   <chr>
#> 1 ""   
#> 2 ""   
#> 3 "a"
```

**Comment by cbrnr** [comment]
I like the Tidyverse way, I didn't know that! Whatever Polars ends up deciding, I think it's also be important to have consistent repr and html_repr, which is currently not the case.

**Comment by cbrnr** [comment]
Note that by "empty" strings the Tidyverse really means "whitespace-only" strings, so this also adds quotes:

```r
> tibble::tibble(a = c(" ", "b", "c"))
#> # A tibble: 3 × 1
#>   a    
#>   <chr>
#> 1 " "  
#> 2 "b"  
#> 3 "c" 
```

Furthermore, the quotes are printed in gray (and not in black) so that they don't distract that much, which I also like!

**Comment by mcrumiller** [comment]
If quotes are used, I think that single quotes look a bit better when *reading* strings. This is how python shows strings:

```pycon
>>> "hello"
'hello'
```

**Comment by liufeimath** [comment]
> If quotes are used, I think that single quotes look a bit better when _reading_ strings. This is how python shows strings:
> 
> ```
> >>> "hello"
> 'hello'
> ```

Yeah that's just what python chose as the default display, which many people hate (and is not consistent with almost every major programming languages out there). I think Polars uses double quotes pretty much every where and that's what Polars choose. I think it's a right choice. Pyarrow also does so (double quote). The default option of popular code formatters (ruff, black) also go with double quote. But taking a step back, I still want to emphasize on consistency, if polars choose single-quote, it's fine, but it needs to be single-quote everywhere (displays, python manuals, documentations, user guides, etc). Mixing quote styles is a bad habit imo.

**Comment by cbrnr** [comment]
I think displaying items without quotes is much more readable, consistent with the regular repr, and consistent with other dataframe implementations (pandas, DataFrames.jl, data.frame, Tibble). Disambiguating a numeric from a str column is still pretty easy because the data type of each column is shown. And if that's not enough, why not go the Tibble route and only show the quotes if there's at least one whitespace-only string item in the column?

**Comment by mcrumiller** [comment]
I believe the issue is that it's hard to tell if a string has whitespace at the edges. ` abc` and `abc` will both look the same.

One option is as you suggest there, but perhaps only use quotes if the string has whitespace at the edges.

Another option is to use `·` to render whitespace, which is fairly common, and which I think may work best, as in:

A          | B
-----------|------------------
`hi·there` | `··space·on·left`
`nope`     | `test`

**Comment by cbrnr** [comment]
Yes, why not, then quotes are never needed I guess? I'd still follow what other packages have already implemented, the Tibble solution is both unintrusive and seems to work well.

**Comment by liufeimath** [comment]
> Disambiguating a numeric from a str column is still pretty easy because the data type of each column is shown.

By the same logic I can say "we should remove the .0 in the float number print output since the type already says it's float, not int, and the extra .0 is just noisy". This is not a good argument, the type in the header info is not a "local info", and you don't expect user to figure that out with the necessity to move eyeball ten lines above to look for type info in the header. Same reason goes to why the "public:/private:" specifier in c++ is very bad, it's just not local enough, and costing reader tens or hundreds of line moves to look for the "environment variable" to figure out the attribute of an entity. In contrast, Rust is much better since you can tell the pub/pri attribute simply by the `pub` keyword presence before the function declaration. That's a bit digress but the same argument applies here as well, if I can tell by the quote that it's a string not a int, why forcing the user to look at the header, which could be many lines above? A string is a string, an int is an int, a float is a float. Let it be self-document enough. Let alone all the spacing issues mentioned above.

But again, taking a step back, I think it's crucial to be consistent, which is way more important than my argument above. One way or the other, we need to pick one. The quoting inconsistency across str and html formats does need a fix.

**Comment by cbrnr** [comment]
Yes, and then fix should be no quotes.

**Comment by hoechenberger** [comment]
Whatever the decision (quotes or no quotes), I'd appreciate it if it was consistent between HTML and string repr :)

**Comment by agossard** [comment]
Could this be a configuration setting the you can toggle with pl.Config, which already controls other aspects of how a dataframe is represented in Jupyter?

**Comment by deanm0000** [comment]
I think there's a case to keep them inconsistent between str and html. The str representation is monospaced so having 2 quotes takes up a lot more space on the screen than the html version so it's not so cumbersome to have quotes in the html representation. Just to flag this...if the str representation gets changed then the `pl.from_repr` function needs to be updated.

**Comment by liufeimath** [comment]
> I think there's a case to keep them inconsistent between str and html. The str representation is monospaced so having 2 quotes takes up a lot more space on the screen than the html version so it's not so cumbersome to have quotes in the html representation. Just to flag this...if the str representation gets changed then the `pl.from_repr` function needs to be updated.

This is a really weak argument to defend the inconsistency.

I agree that `pl.from_repr` also needs a change. My suggestion is that the Series/DataFrame str representations should emulate what the html representations do, where Series is just treated as a 1-column DataFrame with minimal modifications (the shape-tuple)(code pointers [here](https://github.com/pola-rs/polars/blob/811149398ccfd089b775c46a347aac791e24faee/py-polars/polars/series/series.py#L1449) and [here](https://github.com/pola-rs/polars/blob/811149398ccfd089b775c46a347aac791e24faee/py-polars/polars/dataframe/_html.py#L137)). I mentioned this in [another issue](https://github.com/pola-rs/polars/issues/18173#issuecomment-2288609353) related to this one. Overall I think it's much easier if we can just unify the representation styles and Series/DataFrame logic treatments, and it should help both in terms of consistency and beauty, as well as easier code maintenance.

**Comment by deanm0000** [comment]
I'm certainly biased towards the status quo, I won't deny that. If you erased my memory then I can't imagine defending inconsistency. While I acknowledge that, I like the quotes in vsc and I like being able to do print and get the monospacing output without quotes. If you told me I had to choose both with quotes or both without I don't know which I'd prefer.

**Comment by kodonnell** [comment]
For what it's worth, this is 'unexpected' behaviour in my experience with data tools. So much so that my immediate assumption (as a new polars user) was that the CSV I was reading was quote delimited, and polars wasn't smart enough to handle that = ) Wasted 5 minutes trying to force read_csv to properly parse quotes, before realising it was just a display issue! 

I can further imagine if some fields in my CSV started/ended in `"` (but fields aren't delimited) then seeing something like `"my field""` would get extra confusing. (I tested this and it does happen by the looks.) So, worth keeping this angle in mind, regardless of what the "right" way to do it is.

That said, responding to some of the above points:
- Whenever I see a string column, I always assume it may be empty or null (or truncated, or incorrectly encoded, etc. etc.). I don't need quotation marks (or whitespace representations) to tell me that. And given I'm usually only looking at the first 10 rows of a million row dataframe (and you might only have a single row with a space etc.) then displaying in a way to indicate "there might be spaces" is not adding any value at all. With the current quoting, things are a lot harder to read - both extra characters I'm not used to, and that my tables are often hundreds of columns wide, not 3 columns (with simple strings) as used in these examples i.e. screen real estate is precious. So, in my opinion, there's pain, but no gain.
- Likewise for the above case of differentiating between `"1"` and `1`. Parsing strings to numbers is a lot messier than this and if the user's "only" check that their parsing has worked is a visual representation as a `"1"` vs a `1` then they'll likely be surprised in future = ) You also very quickly figure it out as soon as you do any numeric operation on your field, so again, I'm not sure there's much value added.

Given the above, the industry "standard" (?), and the polars string representation doesn't use quotes anyway ... it kinda seems like an obvious choice to me. But I might not be representative = ) 

The Tidyverse approach is a separate-ish discussion I think i.e. I'd worry about row-specific rendering after sorting the generic dataframe rendering. That said, it has similar downsides to above as well (e.g. `" "` vs `"" ""`), so I'd prefer consistency and not a row-specific rendering.
